.specs/unsynced-covalues-tracking/requirements.md

```diff
@@ -0,0 +1,66 @@
+# Track Unsynced CoValues & Resume Sync
+
+## Introduction
+
+Currently, Jazz only resumes sync on CoValues that are loaded in memory. This creates a problem when users partially upload a CoValue graph, go offline, restart the app and then come back online. In this scenario, only the loaded parts of the CoValue graph resume syncing, leaving unloaded parts unsynced even though they may have pending changes.
+
+This feature will implement automatic tracking and resumption of sync for all CoValues with pending changes, regardless of whether they are currently loaded in memory. The solution must work across all platforms (web, React Native, Node.js, cloud workers) and be performant enough to run in cloud environments.
+
+Tracking unsynced CoValues also allows providing reactive APIs for tracking if CoValues have been synced. This includes refactoring `waitForSync` so that we don't need to create a subscription, and adding new subscription APIs for monitoring sync state at different levels of granularity: `CoValueCore.subscribeToSyncStatus` and `SyncManager.subscribeToSyncStatus`.
+
+## User Stories
+
+### US1: Track Unsynced CoValues
+**As a** Jazz user  
+**I want** Jazz to automatically track CoValues that have unsynced changes  
+**So that** these CoValues can be synced even if they're not currently loaded in memory
+
+**Acceptance Criteria:**
+- [ ] When a CoValue has local changes that haven't been fully uploaded to at least one peer, it is tracked as unsynced
+- [ ] When a CoValue becomes fully synced to all peers, it is removed from the unsynced tracking
+- [ ] The tracking persists across app restarts using platform-appropriate storage
+
+### US2: Resume Sync on App Start
```

Comment:
Also on sign up when the dev configured `sync.when: "signUp"` - [docs](https://jazz.tools/docs/react/key-features/authentication/authentication-states#controlling-sync-for-different-authentication-states)

Reply:
How do you feel about forwarding `when` to the local node? To me the correct behaviour if there are no server peers would be:
- if `when="signup"` (and the user is not signed up), track all coValues as unsynced
- if `when="never"`, not track coValues as unsynced

Reply:
Yes, makes sense.